linode/instancedisk/framework_resource.go

```diff
@@ -0,0 +1,471 @@
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ctx = populateLogAttributes(ctx, state)
+
+	linodeID, id := getLinodeIDAndDiskID(state, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client := r.Meta.Client
+
+	tflog.Trace(ctx, "client.GetInstanceDisk(...)")
+	disk, err := client.GetInstanceDisk(ctx, linodeID, id)
+	if err != nil {
+		if lerr, ok := err.(*linodego.Error); ok && lerr.Code == 404 {
+			resp.Diagnostics.AddWarning(
+				"Disk Not Found",
+				fmt.Sprintf(
+					"%s\nRemoving Disk (%d) of Linode instance (%d) from state because it no longer exists",
+					err.Error(), id, linodeID,
+				),
+			)
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Failed to Find the Linode Instance Disk (%d)", id),
+			err.Error(),
+		)
+		return
+	}
+
+	state.FlattenDisk(disk, false)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *Resource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	tflog.Debug(ctx, "Update "+r.Config.Name)
+
+	var plan, state ResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	ctx = populateLogAttributes(ctx, state)
+
+	updateTimeout, diags := plan.Timeouts.Update(ctx, DefaultVolumeUpdateTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	linodeID, id := getLinodeIDAndDiskID(state, &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, updateTimeout)
+	defer cancel()
+
+	client := r.Meta.Client
+	timeoutSeconds := helper.FrameworkSafeFloat64ToInt(updateTimeout.Seconds(), &resp.Diagnostics)
+	size := helper.FrameworkSafeInt64ToInt(plan.Size.ValueInt64(), &resp.Diagnostics)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if !state.Size.Equal(plan.Size) {
+		if err := handleDiskResize(
+			ctx, client, linodeID, id, size, timeoutSeconds,
+		); err != nil {
+			resp.Diagnostics.AddError(
+				fmt.Sprintf("Failed to Resize Disk %d", id), err.Error(),
+			)
+			return
+		}
+	}
+
+	updateOpts := linodego.InstanceDiskUpdateOptions{}
+	shouldUpdate := false
+
+	if !state.Label.Equal(plan.Label) {
+		updateOpts.Label = plan.Label.ValueString()
+		shouldUpdate = true
+	}
+
+	if shouldUpdate {
+		tflog.Debug(ctx, "client.UpdateInstanceDisk(...)", map[string]any{
+			"options": updateOpts,
+		})
+		if _, err := client.UpdateInstanceDisk(ctx, linodeID, id, updateOpts); err != nil {
+			resp.Diagnostics.AddError(
+				fmt.Sprintf("Failed to Update Disk %d", id), err.Error(),
+			)
+			return
+		}
+		tflog.Debug(ctx, "Disk update event finished")
+	}
+
+	// get latest status of the disk
+	tflog.Trace(ctx, "client.GetInstanceDisk(...)")
+	disk, err := client.GetInstanceDisk(ctx, linodeID, id)
+	resp.Diagnostics.AddError(
+		fmt.Sprintf("Failed to Find the Disk (%d)", id),
+		err.Error(),
```

Comment:
When running the tests i get a seg fault here.
The cause doesn't jump out at me though.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1a60886]

goroutine 3690 [running]:
github.com/linode/terraform-provider-linode/v2/linode/instancedisk.(*Resource).Update(0xc000f214a0, {0x348a670, 0xc0005dd0e0}, {{{{0x3492440, 0xc0007a0360}, {0x2d17c20, 0xc0005ddef0}}, {0x3498308, 0xc000f11400}}, {{{0x3492440, ...}, ...}, ...}, ...}, ...)
	/terraform-provider-linode/linode/instancedisk/framework_resource.go:281 +0x906
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).UpdateResource(0xc000bb2680, {0x348a670, 0xc0005dd0e0}, 0xc0008ff530, 0xc0008ff440)
	/go/1.21.0/pkg/mod/github.com/hashicorp/terraform-plugin-framework@v1.5.0/internal/fwserver/server_updateresource.go:122 +0x70e
```

Reply:
Oops, couldn't remember how could I forget that...